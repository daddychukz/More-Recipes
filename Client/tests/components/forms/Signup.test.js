import React from 'react';
import { mount, shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import render from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ConnectedSignup,
{ PureSignup } from '../../../src/components/forms/SignupForm';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const initialState = {
  auth: {
    isAuthenticated: false,
    user: {}
  },
  user: {}
};
const store = mockStore(initialState);

const props = {
  actions: {
    signUp: sinon.spy(() => new Promise(() => {}))
  },
};

const mountedWrapper = mount(
  <BrowserRouter>
    <ConnectedSignup store={store} />
  </BrowserRouter>
);

const shallowWrapper = shallow(<PureSignup {...props} />);
const state = {
  email: 'chuks@yahoo.com',
  password: 'chuks',
  confirmPassword: 'chuks',
  error: false
};
const event = {
  preventDefault: jest.fn(),
  target: {
    password: 'abcde',
    email: 'chuks@yahoo.com',
  }
};

describe('Signup test actiomns', () => {
  it('renders Signup component without crashing', () => {
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('SignupForm').length).toBe(1);
  });

  it('should match component snapshot', () => {
    const tree = render.create(
      <BrowserRouter >
        <PureSignup {...props} />
      </BrowserRouter>);
    expect(tree).toMatchSnapshot();
  });

  it('calls onChange event', () => {
    sinon.spy(shallowWrapper.instance(), 'onChange');
    shallowWrapper.instance().onChange(event);
    expect(shallowWrapper.instance().onChange.calledOnce).toEqual(true);
  });

  it('calls onSubmit event', () => {
    sinon.spy(shallowWrapper.instance(), 'onSubmit');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onSubmit(event);
    expect(shallowWrapper.instance().onSubmit.calledOnce).toEqual(true);
  });

  it('calls onSubmit event with errors when password mismatch', () => {
    state.confirmPassword = 'chuksy';
    shallowWrapper.setState(state);
    shallowWrapper.instance().onSubmit(event);
    expect(shallowWrapper.state().error).toEqual(true);
  });
});
