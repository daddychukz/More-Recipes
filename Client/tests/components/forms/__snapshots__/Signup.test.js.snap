// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup test actiomns should match component snapshot 1`] = `
<div>
  <form
    id="signup-form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control form-control-lg"
        name="FullName"
        onChange={[Function]}
        placeholder="Full Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control form-control-lg"
        name="UserName"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control form-control-lg"
        id="email"
        name="Email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control form-control-lg"
        id="password"
        name="Password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="form-control form-control-lg"
        name="ConfirmPassword"
        onChange={[Function]}
        placeholder="Confirm Password"
        required={true}
        type="password"
        value=""
      />
    </div>
    <input
      className="btn btn-info btn-block"
      type="submit"
      value="Register"
    />
  </form>
</div>
`;
